File: examples/lambda_passing_test.py
"""Test passing lambdas as arguments"""

def apply_func(x: int, f) -> int:
    return f(x)

def test_lambda_passing():
    # Pass lambda to function
    result1 = apply_func(10, lambda x: x * 2)
    result2 = apply_func(5, lambda x: x + 10)
    
    # Store lambda and pass it
    triple = lambda x: x * 3
    result3 = apply_func(7, triple)
    
    return result1 + result2 + result3

File: examples/lambda_passing_test.rs
use serde_json;
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn apply_func(x: i32, f: serde_json::Value) -> i32 {
    f(x)
}
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_lambda_passing() -> i32 {
    let result1 = apply_func(10, |x| x * 2);
    let result2 = apply_func(5, |x| x + 10);
    let triple = |x| x * 3;
    let result3 = apply_func(7, triple);
    result1 + result2 + result3
}
